ttcn3/v2/syntax/syntax.go

```diff
@@ -306,6 +314,7 @@ func (t *tree) searchLines(pos int) int {
 			j = h
 		}
 	}
+	t.cacheLine = int(i) - 1
```

Comment:
```suggestion
        line := int(i) - 1
        t.cachedLineBegin = t.lines[line]
        t.cachedLineEnd = t.lines[lint+1]
	t.cacheLine = line
```

Reply:
i have tried this but for line 319 its giving me `runtime error: index out of range [1] with length 1` as line array contains only one element.

Reply:
You are right. `t.lines[lint+1]` does not work for the last line.

Maybe we could fix it with a condition? No very elegant, I know:
```go
if line < len(t.lines) {
    t.cachedLineEnd = t.lines[line+1]
} else {
    t.cachedLineEnd = len(t.lines)
}
```